src/main/java/edu/kis/powp/jobs2d/drivers/UsageMonitorObserver.java

```diff
@@ -1,15 +1,16 @@
 package edu.kis.powp.jobs2d.drivers;
 
 import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.drivers.extensions.UsageMonitorExtension;
 import edu.kis.powp.jobs2d.features.DriverFeature;
 import edu.kis.powp.observer.Subscriber;
 
 public class UsageMonitorObserver implements Subscriber {
 
     @Override public void update() {
         Job2dDriver currentDriver = DriverFeature.getDriverManager().getCurrentDriver();
-        if (!(currentDriver instanceof Job2dDriverDecorator)) {
-            DriverFeature.getDriverManager().setCurrentDriver(new Job2dDriverDecorator(currentDriver));
+        if (!(currentDriver instanceof UsageMonitorExtension)) {
+            DriverFeature.getDriverManager().setCurrentDriver(new UsageMonitorExtension());
         }
     }
 }
```

Comment:
I think that this code should be replaced by some management features inside your composite.

Reply:
I think it should be how it is, its part of an observer. not of my composite implementation. I have just changed the name of Job2dDriverDecorator to UsageMonitorExtension and made it fluent to use any base driver.